drivers/tee/optee/optee.c

```diff
@@ -152,10 +152,30 @@ static int param_to_msg_param(const struct tee_param *param, unsigned int num_pa
 	return 0;
 }
 
+static int msg_param_to_tmp_mem(struct tee_param *p, uint32_t attr,
+				const struct optee_msg_param *mp)
+{
+	struct tee_shm *shm;
+
+	p->attr = TEE_PARAM_ATTR_TYPE_MEMREF_INPUT + attr - OPTEE_MSG_ATTR_TYPE_TMEM_INPUT;
+	p->b = mp->u.tmem.size;
+	shm = (struct tee_shm *)(unsigned long)mp->u.tmem.shm_ref;
```

Comment:
why do you need two explicit casts in a row ?

Reply:
updated